test/module/irohad/ordering/on_demand_os_test.cpp

```diff
@@ -343,3 +343,21 @@ TEST_F(OnDemandOsTest, SeveralTransactionsOneCommited) {
   // already processed transaction is no present in the proposal
   EXPECT_TRUE(std::find(txs.begin(), txs.end(), batch2_tx) == txs.end());
 }
+
+/**
+ * @given initialized on-demand OS with a batch in collection
+ * @when the same batch arrives, round is closed, proposal is requested
+ * @then the proposal contains the batch once
+ */
+TEST_F(OnDemandOsTest, DuplicateTxTest) {
+  auto now = iroha::time::now();
+  auto txs1 = generateTransactions({1, 2}, now);
+  os->onBatches(txs1);
+
+  auto txs2 = generateTransactions({1, 2}, now);
+  os->onBatches(txs2);
+  os->onCollaborationOutcome(commit_round);
+  auto proposal = os->onRequestProposal(target_round);
+
+  ASSERT_EQ(1, boost::size((*proposal)->transactions()));
+}
```

Comment:
I think better to add/updated existing test for simulator and check block invalidation.

Reply:
It it a private field in Simulator, which is used internally and not accessible with API of Simulator. Could you suggest how to make the test?